varats/varats/plots/blame_interaction_degree.py

```diff
@@ -744,35 +791,73 @@ class BlameLibraryInteraction(Plot):
         """Plot the current plot to a file."""
 
-    def _get_interaction_data(self) -> pd.DataFrame:
+    def _get_interaction_data(self, blame_diff: bool = False) -> pd.DataFrame:
         commit_map: CommitMap = self.plot_kwargs['get_cmap']()
         case_study = self.plot_kwargs.get('plot_case_study', None)
         project_name = self.plot_kwargs["project"]
-        lib_interaction_df = \
-            BlameLibraryInteractionsDatabase.get_data_for_project(
-                project_name, ["revision", "time_id", "base_hash", "base_lib",
-                               "inter_hash", "inter_lib", "amount"],
-                commit_map, case_study)
+
+        variables = [
+            "time_id", "base_hash", "base_lib", "inter_hash", "inter_lib",
+            "amount"
+        ]
+
+        if blame_diff:
+            lib_interaction_df = \
+                BlameDiffLibraryInteractionDatabase.get_data_for_project(
+                    project_name, ["revision", *variables], commit_map,
+                    case_study
+                )
+        else:
+            lib_interaction_df = \
+                BlameLibraryInteractionsDatabase.get_data_for_project(
+                    project_name, ["revision", *variables], commit_map,
+                    case_study
+                )
 
         length = len(np.unique(lib_interaction_df['revision']))
         is_empty = lib_interaction_df.empty
 
-        if is_empty or length == 1:
+        if not blame_diff and (is_empty or length == 1):
             # Need more than one data point
             raise PlotDataEmpty
         return lib_interaction_df
 
     def _graphviz_plot(
         self,
         view_mode: bool,
+        show_blame_interactions: bool = True,
+        show_blame_diff: bool = False,
         show_edge_weight: bool = False,
         shown_revision_length: int = 10,
         edge_weight_threshold: tp.Optional[EdgeWeightThreshold] = None,
+        layout_engine: str = 'fdp',
         save_path: tp.Optional[Path] = None,
-        filetype: str = 'pdf'
+        filetype: str = 'png'
     ) -> tp.Optional[Digraph]:
 
-        df = self._get_interaction_data()
+        def _get_graphviz_project_data(
+            blame_interactions: bool, blame_diff: bool
+        ) -> pd.DataFrame:
+            if not blame_interactions and not blame_diff:
+                LOG.warning(
+                    "You have to set either 'show_blame_interactions', "
+                    "'show_blame_diff', or both to 'True'. Aborting."
+                )
+                raise PlotDataEmpty
+
+            if blame_interactions and blame_diff:
+                inter_df = self._get_interaction_data(False)
+                diff_df = self._get_interaction_data(True)
+                return _add_diff_amount_col_to_df(inter_df, diff_df)
+
+            if blame_interactions:
+                return self._get_interaction_data(False)
+
+            return self._get_interaction_data(True)
```

Comment:
```suggestion
            if blame_interactions:
                inter_df = self._get_interaction_data(False)
            else:
                inter_df = self._get_interaction_data(True)
                
            if  blame_diff:
                diff_df = self._get_interaction_data(True)
                return _add_diff_amount_col_to_df(inter_df, diff_df)

            return inter_df
```
I think this could make it a bit more readable by separating the two cases